var/spack/repos/builtin/packages/hpctoolkit/package.py

```diff
@@ -119,20 +120,30 @@ class Hpctoolkit(AutotoolsPackage, MesonPackage):
         "python", default=False, description="Support unwinding Python source.", when="@2023.03:"
     )
 
-    build_system(conditional("meson", when="@develop"), "autotools", default="autotools")
+    build_system(
+        conditional("meson", when="@2024.01:"),
+        conditional("autotools", when="@:2024.01"),
+        default="autotools",
+    )
 
-    with when("@develop build_system=autotools"):
+    with when("@2024.01: build_system=autotools"):
         depends_on("autoconf", type="build")
         depends_on("automake", type="build")
         depends_on("libtool", type="build")
 
     with when("build_system=meson"):
         depends_on("meson@1.1.0:", type="build")
-        depends_on("gmake", type="build")
-        depends_on("m4", type="build")
-        depends_on("autoconf", type="build")
-        depends_on("automake", type="build")
-        depends_on("libtool", type="build")
+
+        with when("@:2024.01"):
+            depends_on("gmake", type="build")
+            depends_on("m4", type="build")
+            depends_on("autoconf", type="build")
+            depends_on("automake", type="build")
+            depends_on("libtool", type="build")
+
+        with when("@2024.02:"):
+            depends_on("pkgconf", type="build")
+            depends_on("cmake", type="build")
```

Comment:
For my own knowledge, why is `cmake` needed in a `meson` package?

Reply:
Meson prefers not to implement its own dependency search logic, and instead "reuse" the conventions already widely used. In most cases, this means dependencies are found via either `pkg-config` or `cmake` ([official docs](https://mesonbuild.com/Dependencies.html#dependency-detection-method)). The latter is especially handy for libraries that don't ship scripts for either (i.e. no `<lib>.pc` or `<Lib>Config.cmake`), since you can easily fill in the gap with a custom `Find<Lib>.cmake` script.

In our case, we require it to find `intel-tbb` and `dyninst`, which only ship `*Config.cmake` scripts. We also use custom `Find*.cmake` scripts for `intel-xed`, `libpfm4`, `libiberty` and some ROCm tools (`rocprofiler-dev` and `roctracer-dev`).